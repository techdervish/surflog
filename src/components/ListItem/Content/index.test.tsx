import {
  expectWrappersTreeToMatchSnapshot,
  createShallowWrapper,
} from 'utils/testHelpers';
import Component from './index';

const createWrapper = createShallowWrapper(Component);

describe('Content component', () => {
  const wrapper = createWrapper({
    record: { content: 'Record content' },
    occurence: { position: [0, 5] },
  });

  it('renders the expected snapshot', () => {
    expectWrappersTreeToMatchSnapshot(wrapper);
  });

  it('renders just content if there is no match', () => {
    const wrapper = createWrapper({
      record: { content: 'Record content' },
    });
    expect(
      wrapper
        .find('div')
        .first()
        .text(),
    ).toEqual('Record content...');
  });

  it('renders match there is a match', () => {
    const wrapper = createWrapper({
      record: { content: 'Record content' },
      occurence: { position: [0, 5] },
    });
    expect(wrapper.find('span')).toHaveLength(1);
  });
});
