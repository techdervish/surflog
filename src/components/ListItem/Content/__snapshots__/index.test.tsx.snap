// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content component renders the expected snapshot 1`] = `
<div
  key="match_undefined"
>
  <span
    style={
      Object {
        "background": "yellow",
      }
    }
  >
    Recor
  </span>
  d content
  ...
</div>
`;
